geoh5py/objects/points.py

```diff
@@ -31,76 +31,68 @@ class Points(ObjectBase):
     Points object made up of vertices.
     """
 
-    __TYPE_UID = uuid.UUID("{202C5DB1-A56D-4004-9CAD-BAAFD8899406}")
+    _TYPE_UID: uuid.UUID | None = uuid.UUID("{202C5DB1-A56D-4004-9CAD-BAAFD8899406}")
+    __VERTICES_DTYPE = np.dtype([("x", "<f8"), ("y", "<f8"), ("z", "<f8")])
+    _default_name = "Points"
 
-    def __init__(self, object_type: ObjectType, name="Points", **kwargs):
-        self._vertices: np.ndarray | None = None
+    def __init__(
+        self,
+        object_type: ObjectType,
+        vertices: np.ndarray | list | tuple | None = None,
+        **kwargs,
+    ):
+        self._vertices: np.ndarray = self.validate_vertices(vertices)
 
-        super().__init__(object_type, name=name, **kwargs)
+        super().__init__(object_type, **kwargs)
 
-    @classmethod
-    def default_type_uid(cls) -> uuid.UUID:
-        return cls.__TYPE_UID
+    def copy(
+        self,
+        parent=None,
+        copy_children: bool = True,
+        clear_cache: bool = False,
+        mask: np.ndarray | None = None,
+        **kwargs,
+    ) -> Points:
+        """
+        Sub-class extension of :func:`~geoh5py.shared.entity.Entity.copy`.
+        """
+        if mask is not None and self.n_vertices:
+            if not isinstance(mask, np.ndarray) or mask.shape != (self.n_vertices,):
+                raise ValueError("Mask must be an array of shape (n_vertices,).")
+
+            kwargs.update({"vertices": self.vertices[mask]})
+
+        new_entity = super().copy(
+            parent=parent,
+            copy_children=copy_children,
+            clear_cache=clear_cache,
+            mask=mask,
+            **kwargs,
+        )
+
+        return new_entity
 
     @property
-    def extent(self) -> np.ndarray | None:
+    def extent(self) -> np.ndarray:
```

Comment:
Same here, it's a bit more complicated because not all objects have an extent, but all Points and Grids sub-classes do. I would rather keep those classes always returning an array rather than always checking for None

Reply:
Ok I managed to move some of it.